hyperhtml.js

```diff
@@ -193,321 +156,692 @@ var hyperHTML = (function () {'use strict';
           childNodes = slice.call(fragment.childNodes);
-          parentNode.insertBefore(fragment, node);
+          node.parentNode.insertBefore(fragment, node);
           break;
         default:
-          if (Array.isArray(value)) {
-            if (value.length === 0) {
+          if (isArray(value)) {
+            var length = value.length;
+            if (length === 0) {
               removeNodeList(childNodes, 0);
               childNodes = [];
-            } else if(typeof value[0] === 'string') {
-              any(value.join(''));
             } else {
-              i = indexOfDifferences(childNodes, value);
-              if (-1 < i) {
-                removeNodeList(childNodes, i);
-                value = value.slice(i);
-                appendNodes(fragment, value);
-                parentNode.insertBefore(fragment, node);
-                childNodes = childNodes.slice(0, i).concat(value);
+              var type = typeof value[0];
+              if (type === 'string') {
+                any(value.join(''));
+              } else {
+                var i = indexOfDifferences(childNodes, value);
+                if (-1 < i) {
+                  var fragment = emptyFragment(node);
+                  removeNodeList(childNodes, i);
+                  value = value.slice(i);
+                  appendNodes(fragment, value);
+                  node.parentNode.insertBefore(fragment, node);
+                  childNodes = childNodes.slice(0, i).concat(value);
+                }
               }
             }
           } else {
             removeNodeList(childNodes, 0);
             childNodes = value.nodeType === DOCUMENT_FRAGMENT_NODE ?
               slice.call(value.childNodes) :
               [value];
-            parentNode.insertBefore(value, node);
+            node.parentNode.insertBefore(value, node);
           }
           break;
       }
     };
   }
 
-  // basic closure to update nodes textContent
-  //
-  // render`
-  //  <p>
-  //    ${'spaces around means textContent'}
-  //  </p>`;
-  function setTextContent(node) {
-    var oldValue;
-    return function text(newValue) {
-      if (oldValue !== newValue) {
-        node.textContent = (oldValue = newValue);
+  // ---------------------------------------------
+  // DOM Traversing
+  // ---------------------------------------------
+
+  // look for attributes that contains the comment text
+  function attributesSeeker(node, paths) {
+    for (var
+      attribute,
+      value = IE ? UID : UIDC,
+      attributes = node.attributes,
+      i = 0, length = attributes.length;
+      i < length; i++
+    ) {
+      attribute = attributes[i];
+      if (attribute.value === value) {
+        paths.push(
+          Path(
+            'attr',
+            // with IE the order doesn't really matter
+            // as long as the right attribute is addressed
+            IE ?
+              node[GET_ATTRIBUTE_NODE](IEAttributes.shift()) :
```

Comment:
IEAttributes is populated while replacing comments in the RegExp.

The discovery order will NOT match the attributes order here, which is why it has to be drained to get the right attribute left to right.

I give you an example. `<div attr="${1}" whatever="0" id="${2}">` in IE will result into an attributes list of ['id', 'attr'] and it's unpredictable with `class` and any other possible special attribute per node.

The RegExp will replace, left to right, the `<div attr=<!--UID-->" whatever="0" id="<!--UID-->">` as tring, because IE also crashes if it has attributes as comment (Edge inherited the shuffled order issue but at least not the throwing error one so it needs this hack too).

At that point it doesn't matter which attribute, in which order, the UID content is discovered, what we want to map in there is the attribute node `attr` first and the `id` after.

TL;DR it's a hack that resolve IE problems and it's granted to work thanks to tests.

Yes, it is scary, because scary has always been IE in general.